content/academy/puppeteer_playwright/collecting_data.md

```diff
@@ -0,0 +1,284 @@
+---
+title: IV - Collecting data
+description: Learn how to collect data from a page with evaluate functions, then how to safely collect it by using a second library called Cheerio.
+menuWeight: 7.4
+paths:
+    - puppeteer-playwright/collecting-data
+---
+
+# [](#collecting-data) Collecting data
+
+Now that we know how to execute scripts on a page, we're ready to learn a bit about [data collection]({{@link web_scraping_for_beginners/data_collection.md}}). In this lesson, we'll be scraping all of the on-sale products from our [Fakestore](https://demo-webstore.apify.org/search/on-sale) website.
+
+> Most data collection cases involve looping through a list of items of some sort.
+
+There are two main ways to collect data with Playwright and Puppeteer:
+
+1. Directly in `page.evaluate()` and other evaluate functions such as `page.$$eval()`.
+2. In the Node.js context using a parsing library such as [Cheerio](https://www.npmjs.com/package/cheerio)
+
+<!-- Understanding the two different contexts your code is running is  -->
+
+## Setup
+
+Here is the base set up for our code, upon which we'll be building off of in this lesson:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+const page = await browser.newPage();
+
+await page.goto('https://demo-webstore.apify.org/search/on-sale');
+
+await page.waitForLoadState('networkidle');
```

Comment:
You can add this as an option to `page.goto`. Also, is this really needed? The best practice is to wait for the page to `load` or sometimes even ``domcontentloaded` to speed up loading and then wait for the specific elements you need separately. This should also be part of the "Waiting for content" lesson.